controllers/k8ssandra/datacenters.go

```diff
@@ -52,7 +53,7 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 	}
 
 	// Reconcile CassandraDatacenter objects only
-	for idx, dcTemplate := range kc.Spec.Cassandra.Datacenters {
+	for idx, dcTemplate := range sortDatacentersByPriority(kc.Spec.Cassandra.Datacenters) {
```

Comment:
Open question: this means that dcs will always be reconciled in this particular order, even if no upgrades are happening. IOW, when creating a new DSE cluster, we would first create the Analytics DCs, then the Cassandra/Graph ones, then the Search ones. But is it OK to bootstrap an Analytics DC before the Cassandra ones?